src/library/export/trackexport.cpp

```diff
@@ -0,0 +1,129 @@
+#include "library/export/trackexport.h"
+
+#include <QFileInfo>
+#include <QMessageBox>
+#include <QDebug>
+
+void TrackExport::run() {
+    int i = 0;
+    for (const auto& track : m_tracks) {
+        QString path = track->getLocation();
+        QFileInfo fileinfo(path);
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
+        exportTrack(track);
+        if (m_bStop) {
+            emit(canceled());
+            return;
+        }
+        ++i;
+        emit(progress(fileinfo.fileName(), i, m_tracks.size()));
+    }
+}
+
+void TrackExport::exportTrack(TrackPointer track) {
+    QString sourceFilename = track->getLocation();
+    QFileInfo source_fileinfo(sourceFilename);
```

Comment:
Best to re-use QFileInfo's if we can since each one results in a stat operation. Could this method take the QFileInfo you already created in run()?